### PR TITLE
feat: support ttheader streaming codec

### PR DESCRIPTION
**Pull request by joway**
#### What type of PR is this?

feat

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [ ] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional):

### CODE REVIEW

File: bufiox/defaultbuf.go [hunk 1]
```diff
@@ -43,18 +44,30 @@ const (
...
-func NewBytesReader(buf []byte) *BytesReader {
-	r := &BytesReader{}
+func NewBytesReader(buf []byte) (r *BytesReader) {
+	v := bytesReaderPool.Get()
+	if v == nil {
+		r = &BytesReader{}
+	} else {
+		r = v.(*BytesReader)
+	}
 	r.reset(r.fakedIOReader, buf)
```
Comment (jayantxie) [inline bufiox/defaultbuf.go:71]: r.fakedIOReader = nil

File: bufiox/defaultbuf.go [hunk 2]
```diff
@@ -208,6 +227,8 @@ func (r *DefaultReader) Release(e error) error {
 		}
 	}
 	r.ri = 0
+	r.err = nil
+	defaultReaderPool.Put(r)
```
Comment (jayantxie) [inline bufiox/defaultbuf.go:231]: release不能释放对象的，这个要长期持有

Comment (jayantxie) [reply bufiox/defaultbuf.go:231]: release语义是把已经读完的buffer释放掉，但未读的buffer仍然是可用的。注释里面可能有歧义，得微调一下

Comment (joway) [reply bufiox/defaultbuf.go]: @jayantxie revert 了这个性能优化的commit，再看看呢

**Review by jayantxie — APPROVED** [review]

**Pull request merged**